From cff5747c74c41b22f1ce1340978b1c226a8cdf32 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Wed, 13 Jul 2016 15:04:30 -0400
Subject: [PATCH] fix regression in tcsetattr on all mips archs

revert commit 8c316e9e49d37ad92c2e7493e16166a2afca419f. it was wrong
and does not match how the kernel API works.
---
 arch/mips/bits/termios.h    | 6 +++---
 arch/mips64/bits/termios.h  | 6 +++---
 arch/mipsn32/bits/termios.h | 6 +++---
 4 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/arch/mips/bits/termios.h b/arch/mips/bits/termios.h
index f559f76..6a1205d 100644
--- a/arch/mips/bits/termios.h
+++ b/arch/mips/bits/termios.h
@@ -141,9 +141,9 @@ struct termios {
 #define TCOFLUSH  1
 #define TCIOFLUSH 2
 
-#define TCSANOW 0x540e
-#define TCSADRAIN 0x540f
-#define TCSAFLUSH 0x5410
+#define TCSANOW   0
+#define TCSADRAIN 1
+#define TCSAFLUSH 2
 
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 #define EXTA    0000016
diff --git a/arch/mips64/bits/termios.h b/arch/mips64/bits/termios.h
index f559f76..6a1205d 100644
--- a/arch/mips64/bits/termios.h
+++ b/arch/mips64/bits/termios.h
@@ -141,9 +141,9 @@ struct termios {
 #define TCOFLUSH  1
 #define TCIOFLUSH 2
 
-#define TCSANOW 0x540e
-#define TCSADRAIN 0x540f
-#define TCSAFLUSH 0x5410
+#define TCSANOW   0
+#define TCSADRAIN 1
+#define TCSAFLUSH 2
 
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 #define EXTA    0000016
diff --git a/arch/mipsn32/bits/termios.h b/arch/mipsn32/bits/termios.h
index f559f76..6a1205d 100644
--- a/arch/mipsn32/bits/termios.h
+++ b/arch/mipsn32/bits/termios.h
@@ -141,9 +141,9 @@ struct termios {
 #define TCOFLUSH  1
 #define TCIOFLUSH 2
 
-#define TCSANOW 0x540e
-#define TCSADRAIN 0x540f
-#define TCSAFLUSH 0x5410
+#define TCSANOW   0
+#define TCSADRAIN 1
+#define TCSAFLUSH 2
 
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 #define EXTA    0000016
-- 
2.8.1

